[SPARK-15467][build] update janino version to 3.0.0

kiszk opened the pull request:
## What changes were proposed in this pull request?

This PR updates version of Janino compiler from 2.7.8 to 3.0.0. This version fixes [an Janino issue](https://github.com/janino-compiler/janino/issues/1) that fixes [an issue](https://issues.apache.org/jira/browse/SPARK-15467), which throws Java exception, in Spark.
## How was this patch tested?

Manually tested using a program in [the JIRA entry](https://issues.apache.org/jira/browse/SPARK-15467)

kiszk committed:
update janino version to 3.0.0

SparkQA commented:
**[Test build #62060 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62060/consoleFull)** for PR 14127 at commit [`ce31dda`](https://github.com/apache/spark/commit/ce31ddade28039aafceb7ab1516b8f79bd2948f1).

SparkQA commented:
**[Test build #62060 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62060/consoleFull)** for PR 14127 at commit [`ce31dda`](https://github.com/apache/spark/commit/ce31ddade28039aafceb7ab1516b8f79bd2948f1).
 * This patch **fails build dependency tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62060/
Test FAILed.

srowen commented:
You'll need to update the deps files. More importantly what are the potential incompatible changes that could affect Spark?

kiszk committed:
update deps files

SparkQA commented:
**[Test build #62063 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62063/consoleFull)** for PR 14127 at commit [`2c702c5`](https://github.com/apache/spark/commit/2c702c5f24f29ef8b8e55160e95fc97044ec7276).

kiszk commented:
[Here](http://janino-compiler.github.io/janino/changelog.html) is a change log of Janino. Updates from 2.7.8 are the followings:

- JANINO-186: Code size error message should include method name: Also report when the constant pool size limit is exceeded.
- JANINO-185: Improve logging: Replaced good ol' `if (DEBUG) System.err.println(...)` with `java.util.logging{.
- `org.codehaus.janino.ScriptEvaluator` now supports the concept of "local methods".
- Implemented annotations (declarations and uses)
- Implemented enums (uses only)
- Added (limited) support for method type parameters.
- Added support for annotations on package declarations.
- Added an "architectural overview" in the JAVADOC.

The first item is a fix for the reported issue.  Other items seem to be to add new features. While I think that there is no potential incompatible changes according to changelog, I would like to see the result of Jenkins.

SparkQA commented:
**[Test build #62063 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62063/consoleFull)** for PR 14127 at commit [`2c702c5`](https://github.com/apache/spark/commit/2c702c5f24f29ef8b8e55160e95fc97044ec7276).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62063/
Test PASSed.

rxin commented:
Thanks - merging in master.

srowen commented:
Sounds like a small set of changes; not sure why it was a major version change. _shrug_
